tests/run_tests.sh

```diff
@@ -0,0 +1,247 @@
+    FILE_LOG_LEVEL='debug'
+    SKIP_FOLDERS="ptftests acstests saitests"
+    EXTRA_PARAMETERS=""
+    TEST_CASES=""
+    OMIT_FILE_LOG="False"
+    BYPASS_UTIL="False"
+    RETAIN_SUCCESS_LOG="False"
+
+    TEST_METHOD='group'
+
+    TESTBED_FILE="${BASE_PATH}/ansible/testbed.csv"
+    INVENTORY="${BASE_PATH}/ansible/lab,${BASE_PATH}/ansible/veos"
+
+    export ANSIBLE_CONFIG=${BASE_PATH}/ansible
+    export ANSIBLE_LIBRARY=${BASE_PATH}/ansible/library/
+} 
+
+function setup_test_options()
+{
+    PYTEST_COMMON_OPTS="--inventory ${INVENTORY} \
+                      --host-pattern ${DUT_NAME} \
+                      --testbed ${TESTBED_NAME} \
+                      --testbed_file ${TESTBED_FILE} \
+                      --log-cli-level ${CLI_LOG_LEVEL} \
+                      --log-file-level ${FILE_LOG_LEVEL} \
+                      --allow_recover \
+                      --showlocals \
+                      --assert plain \
+                      -rav"
+
+    for skip in ${SKIP_SCRIPTS} ${SKIP_FOLDERS}; do
+        PYTEST_COMMON_OPTS="${PYTEST_COMMON_OPTS} --ignore=${skip}"
+    done
+
+    rm -rf logs
+    if [[ x"${OMIT_FILE_LOG}" == x"True" ]]; then
+        UTIL_LOGGING_OPTIONS=""
+        TEST_LOGGING_OPTIONS=""
+    else
+        mkdir logs
+
+        UTIL_LOGGING_OPTIONS="--junit-xml=logs/util.xml --log-file=logs/util.log"
+        TEST_LOGGING_OPTIONS="--junit-xml=logs/tr.xml --log-file=logs/test.log"
+    fi
+    UTIL_TOPOLOGY_OPTIONS="--topology util"
+    TEST_TOPOLOGY_OPTIONS="--topology ${TOPOLOGY}"
+}
+
+function run_debug_tests()
+{
+    echo "=== Show test settings ==="
+    echo "SCRIPT:                ${SCRIPT}"
+    echo "FULL_PATH:             ${FULL_PATH}"
+    echo "SCRIPT_PATH:           ${SCRIPT_PATH}"
+    echo "BASE_PATH:             ${BASE_PATH}"
+
+    echo "ANSIBLE_CONFIG:        ${ANSIBLE_CONFIG}"
+    echo "ANSIBLE_LIBRARY:       ${ANSIBLE_LIBRARY}"
+    echo "BYPASS_UTIL:           ${BYPASS_UTIL}"
+    echo "CLI_LOG_LEVEL:         ${CLI_LOG_LEVEL}"
+    echo "EXTRA_PARAMETERS:      ${EXTRA_PARAMETERS}"
+    echo "FILE_LOG_LEVEL:        ${FILE_LOG_LEVEL}"
+    echo "INVENTORY:             ${INVENTORY}"
+    echo "OMIT_FILE_LOG:         ${OMIT_FILE_LOG}"
+    echo "RETAIN_SUCCESS_LOG:    ${RETAIN_SUCCESS_LOG}"
+    echo "SKIP_SCRIPTS:          ${SKIP_SCRIPTS}"
+    echo "SKIP_FOLDERS:          ${SKIP_FOLDERS}"
+    echo "TEST_CASES:            ${TEST_CASES}"
+    echo "TEST_METHOD:           ${TEST_METHOD}"
+    echo "TESTBED_FILE:          ${TESTBED_FILE}"
+    echo "TEST_LOGGING_OPTIONS:  ${TEST_LOGGING_OPTIONS}"
+    echo "TEST_TOPOLOGY_OPTIONS: ${TEST_TOPOLOGY_OPTIONS}"
+    echo "UTIL_LOGGING_OPTIONS:  ${UTIL_LOGGING_OPTIONS}"
+    echo "UTIL_TOPOLOGY_OPTIONS: ${UTIL_TOPOLOGY_OPTIONS}"
+
+    echo "PYTEST_COMMON_OPTS:    ${PYTEST_COMMON_OPTS}"
+}
+
+function run_group_tests()
+{
+    if [[ x"${BYPASS_UTIL}" == x"False" ]]; then
+        echo "=== Runing utility test cases ==="
```

Comment:
Could this language be confusing? Should we describe this as "pre-steps" or "test setup" or something similar? I'm just wondering because we're not really testing anything per se (at least I don't think so?), we're preparing the testbed for further steps.

Reply:
Thanks! Changed.